plyflatten/utils.py

```diff
@@ -9,30 +9,36 @@ class InvalidPlyCommentsError(Exception):
     pass
 
 
-def crs_proj(crs_code, crs_type="UTM"):
+def crs_proj(crs_params, crs_type="UTM"):
     """
     Return a pyproj.Proj object that corresponds
-    to the given UTM zone string or EPSG code
+    to the given UTM zone string or the CRS parameters
 
     Args:
-        crs_code (str): UTM zone number + hemisphere (eg: '30N') or EPSG code
-        crs_type (str): 'UTM' (default) or 'EPSG'
+        crs_params (int, str, dict): UTM zone number + hemisphere (eg: '30N') or an authority
+            string (EPSG:xxx) or any other format supported by pyproj
+        crs_type (str): 'UTM' (default) or 'CRS'
 
     Returns:
-        pyproj.Proj: object that can be used to transform coordinates
+        pyproj.Proj or pyproj.crs.CRS: object that can be used to transform coordinates
     """
     if crs_type == "UTM":
-        zone_number = crs_code[:-1]
-        hemisphere = crs_code[-1]
+        zone_number = crs_params[:-1]
+        hemisphere = crs_params[-1]
         return pyproj.Proj(
             proj="utm", zone=zone_number, ellps="WGS84", datum="WGS84", south=(hemisphere == "S")
         )
-    elif crs_type == "EPSG":
-        return pyproj.Proj("epsg:{}".format(crs_code))
+    elif crs_type == "CRS":
+        if isinstance(crs_params, str):
+            try:
+                crs_params = int(crs_params)
+            except (ValueError, TypeError):
+                pass
+        return pyproj.crs.CRS(crs_params)
 
 
 def crs_code_from_comments(comments, crs_type="UTM"):
-    re_type = "[0-9]{1,2}[NS]" if crs_type == "UTM" else "[0-9]{4,5}"
+    re_type = "[0-9]{1,2}[NS]" if crs_type == "UTM" else "(.*)"
```

Comment:
I cannot tell you why I simply use the parenthesis because it is working like this :)

Reply:
The final regexes look like:
```
1. ^projection: {} ([0-9]{1,2}[NS])
2. ^projection: {} ((.*))
```

In case 1., you have one capturing group, whereas in case 2., you have a capturing group within another capturing group. It works because when you access `re.search.group(1)` it must refer to the innermost capturing group, but I guess it would be clearer without it (and it works according to my tests locally).

```suggestion
    re_type = "[0-9]{1,2}[NS]" if crs_type == "UTM" else ".*"
```